kubeadm/tasks/prerequisites.yml

```diff
@@ -1,10 +1,13 @@
 ---
 
+# This is needed to fix a metadata issue with yum
+# Something in the Ansible yum module is re-instating the bad metadata
 - name: Ensure all non-Kubernetes packages are at the latest version
-  yum:
-    name: '*'
-    state: latest
-    disablerepo: kubernetes
+  shell: >
+    yum update -y --disablerepo=kubernetes || (
+      yum clean metadata &&
+      yum update -y --disablerepo=kubernetes
+    )
```

Comment:
How about this:
```
    - yum:
        name: '*'
        state: latest
        disablerepo: kubernetes
      register: yum_state
      ignore_errors: true
    - block:
        - command: yum clean metadata
        - yum:
            name: '*'
            state: latest
            disablerepo: kubernetes
      when: yum_state is failed
```

Reply:
@brtknr 

Have you tried this and seen it work? Because I originally did it like this and there seemed to be something in the yum module that was re-exposing the problem in the second invocation. 

I wouldn’t have gone for what I did otherwise!

Reply:
Yep, I tried this and it worked for me